contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -395,6 +395,58 @@ namespace eosiosystem {
    typedef eosio::multi_index< "delband"_n, delegated_bandwidth > del_bandwidth_table;
    typedef eosio::multi_index< "refunds"_n, refund_request >      refunds_table;
 
+   /**
+    * `rex_config` structure within the rex configuration table.
+    * 
+    * @details A rex configuration table entry is defined by:
+    * - `config_id` - configuration id key
+    * - `config_item_name` - configuration item name
+    * - `config_item_value` - configuration value (uint64_t)
+    */
+    struct [[eosio::table,eosio::contract("eosio.system")]] rex_config {
+	uint64_t   	config_id = 0;
+	std::string	config_item_name;
+	uint64_t	config_item_value = 0;
+
+	uint64_t primary_key()const { return config_id; }
+
+	// explicit serialization macro is not necessary, used here only to improve compilation time
+	EOSLIB_SERIALIZE( rex_config, (config_id)(config_item_name)(config_item_value) )
+    };
+
+   /**
+    * Rex configuration table
+    *
+    * @details The rex configuration table is storing only one instance of rex_config which stores
+    * configuration items for the REX system.
+    */
+   typedef eosio::multi_index<"rexconfig"_n, rex_config> rex_config_table;
+
+   /**
+    * `rex_whitelist` structure within the rex whitelisting table.
+    * 
+    * @details A rex whitelist table entry is defined by:
+    * - `whitelist_id` - whitelist id key
+    * - `whitelist_account_name` - configuration item name
+    */
+    struct [[eosio::table,eosio::contract("eosio.system")]] rex_whitelist {
+	uint64_t   	whitelist_id = 0;
+	name		whitelist_account_name;
```

Comment:
whitelist_account_name could be the primary key. This way 50% RAM (if we don't count the RAM needed by the index/row itself) would be saved. Granted, that there probably will not be many whitelisted accounts.

This approach also requires extra CPU, since it will search the DB in Wasm, instead of Boost multi index intrinsics, which would be case with primary keys. When TRIM plan is implemented, there will be no leeway, so we will risk that the DB will grow too large in the future, to be searched through.

Using "name" as the primary key is a common practice, and is used for example in the example contract provided by b1: https://developers.eos.io/eosio-home/docs/data-persistence

Reply:
Yes, I only recently realised how this works - the primary key value is handled automatically by Boost, so it doesn't matter what datatype you use as the return object. Convenient! @craigbranscom has adjusted this in PR #5 (as well as optimized a heap of other code!) 👍